cirq/ops/thermal_relaxation_channel.py

```diff
@@ -0,0 +1,267 @@
+                 p_dephasing: float) -> None:
+        r"""The thermal relaxation channel.
+
+        Construct a channel to model energy dissipation into the environment
+        as well as the environment depositing energy into the system. The
+        probabilities with which the energy exchange occur are given by `gamma`
+        and `beta` with the probability of the environment being not excited is
+        given by `p`.
+
+        The stationary state of this channel is the diagonal density matrix
+        with probability `p` of being |0⟩ and probability `1-p` of being |1⟩.
+
+        Recall the kraus opeartors for the `GeneralizedAmplitudeDampingChannel`:
+
+            $$
+            \begin{aligned}
+            M_0 =& \sqrt{p_exchange} \begin{bmatrix}
+                                1 & 0  \\
+                                0 & \sqrt{1 - p_relaxation}
+                            \end{bmatrix}
+            \\
+            M_1 =& \sqrt{p_exchange} \begin{bmatrix}
+                                0 & \sqrt{p_relaxation} \\
+                                0 & 0
+                           \end{bmatrix}
+            \\
+            M_2 =& \sqrt{1-p_exchange} \begin{bmatrix}
+                                \sqrt{1-p_relaxation} & 0 \\
+                                 0 & 1
+                              \end{bmatrix}
+            \\
+            M_3 =& \sqrt{1-p_exchange} \begin{bmatrix}
+                                 0 & 0 \\
+                                 \sqrt{p_relaxation} & 0
+                             \end{bmatrix}
+            \end{aligned}
+            $$
+
+        This has a choi representation of:
+
+            $$
+             \begin{aligned}
+                CHO =&  \begin{bmatrix}
+                            1 - (1-p_exchange) p_relaxation & 0 & 0 &
+                                \sqrt{1 - p_relaxation} \\
+                            0 & (1-p_exchange) p_relaxation 0 & 0 \\
+                            0 & 0 & p_exchange p_relaxation & 0 \\
+                            \sqrt{1 - p_relaxation} & 0 & 0 &
+                                1 - p_exchange p_relaxation
+                        \end{bmatrix}
+             \end{aligned}
+            $$
+
+        Where each 2x2 block can be thought of as how the channel acts on each
+        basis element in the space of density matrices.
+
+        For the `ThermalRelaxationChannel` channel we simply substitute
+        `p_dephasing` in for `p_relaxation` on the off diagonal elements of the
+        above choi matrix and then evolve the channel using the kraus operators
+        found by sqrt factorization of the choi matrix.
+
+        Args:
+            p_relaxation: the probability of amplitude damping/relaxation
+                occuring.
+            p_dephasing: the probability of dephasing occuring.
+            p_exchange: the probability of the qubit and environment exchanging
+                energy.
```

Comment:
This is unclear. Relaxation is also an example of energy exchange between qubit and environment. Do you mean to imply that `p_exchange` subsumes `p_relaxation`? In this case we must require that `p_exchange > p_relaxation`. I think it'd be cleaner to have API without such dependency on parameters. For example, you could take `p_relaxation` and `p_excitation`.

Reply:
`p_exchange` is still needed because it will eventually give you the population of the excited state or how often you get a 1 when you want a 0 as time grows. I don't think we can do without `p_exchange`, but maybe I can update it to something clearer

Reply:
Isn't it true that `p_exchange = p_relaxation + p_excitation`? If it is, then we can compute p_exchange from the other two. So if we add p_excitation as a parameter then we end up with sufficient information to instantiate the channel, but without the requirement that renders some parameter combinations invalid (such as p_relaxation exceeding p_exchange).

Or am I misunderstanding the meaning of the parameters?